lib/modules/datasource/terraform-module/index.ts

```diff
@@ -80,9 +81,11 @@ export class TerraformModuleDatasource extends TerraformDatasource {
     let pkgUrl: string;
 
     try {
-      // TODO: types (#7154)
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      pkgUrl = `${registryUrl}${serviceDiscovery['modules.v1']}${repository}`;
+      pkgUrl = joinUrlParts(
+        registryUrl,
+        serviceDiscovery['modules.v1']!,
+        repository
+      );
       res = (await this.http.getJson<TerraformRelease>(pkgUrl)).body;
       const returnedName = res.namespace + '/' + res.name + '/' + res.provider;
```

Comment:
this should probably also use `joinUrlParts`
```suggestion
      const returnedName = joinUrlParts(res.namespace, res.name,  res.provider);
```